src/controllers/user.js

```diff
@@ -83,4 +84,44 @@ const logout = (req, res) => {
   }
 };
 
-module.exports = { login, register, logout };
+const getUserProfile = async (req, res, next) => {
+  const { username } = req.params;
+  try {
+    const user = await User.findOne({ username });
+
+    if (!user) return res.status(404).json({ message: 'User not found' });
+
+    const info = {
+      username: user.username,
+      email: user.email,
+      name: user.name,
+      role: user.role,
+      type: user.type,
+    };
+    const donations = [];
+
+    for (let i = 0; i < user.donations.length; i += 1) {
+      const project = Project.findById(user.donations[i].projectID);
```

Comment:
I'm pretty sure you can use populate here too to fetch all the projects

Reply:
hmm i dont think so because here the project in user are the project that they created or managed not the projects supported, the project supported are in donations and they are embedded

Reply:
@felmez well i think you can populate nested fields in our case here we want to populate this:
![image](https://user-images.githubusercontent.com/61846570/152514710-e7313b70-38a8-47d3-8717-4b26eda0fdd8.png)
and there's a way to populate nested fields: https://stackoverflow.com/questions/19222520/populate-nested-array-in-mongoose#:~:text=You%20can%20populate%20multiple%20nested%20documents%20like%20this.

Reply:
ok, let me check this

Reply:
done